flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -202,19 +199,24 @@ public ExecutionVertexID getID() {
     }
 
     public int getNumberOfInputs() {
-        return this.inputEdges.length;
+        return getAllConsumedPartitions().size();
+    }
+
+    public List<ConsumedPartitionGroup> getAllConsumedPartitions() {
```

Comment:
nit: Should we rather call these methods `getAllConsumedPartitionGroups`? Same with `getConsumedPartitionGroup` instead of `getConsumesPartitions`.